### ISSUE TITLE
 OID is not based on criteria

### ISSUE THREAD

**Issue by liwei62** [openssl/openssl#21991]
In OpenSSL, the OID obtained based on the short name "x509" is {0x55,0x04}. However, the OID indicates "Attribute types" in http://www.oid-info.com/get/2.5.4, which may cause unexpected errors in cross-platform certificate parsing.

**Comment by levitte** [comment]
I agree it's a bug, but oh boy this has been with us since the dawn of time...  I'm just going to guess that this was the name used for that arc back in the beginning of X.509 specification, before ITU-T et al formalized it further.

**Comment by t8m** [comment]
What is this oid actually used for?

**Comment by levitte** [comment]
It's the arc for all the attributes in a certificate subject or issuer

**Comment by t8m** [comment]
Yeah, but is this particular oid used alone anywhere? I.e., what is the impact if we rename this to id-at and add a new x509 oid 1.3.6.1.1.15

**Comment by t8m** [comment]
Or, would it be better to just document it is a wrong name for this OID? I do not really think this causes _unexpected errors in cross-platform certificate parsing_. It is just confusing.

**Comment by levitte** [comment]
Both 2.5.4 and 1.3.6.1.1.15 are arcs, so I expect them never to be used directly (I just grepped for `NID_X509`, *we* are certainly not using it directly anywhere).  So the only real danger is if someone wants to add a new leaf to either of those arcs (which they shouldn't) and it becomes an incorrect OID because they were mislead by our incorrect naming.

I for one would opt for correctness insofar that we try to match the rest of the world in our OID naming.

**Comment by t8m** [comment]
OK, then I would treat it as a bug. Not sure whether it should be fixed also on 3.1 and 3.0 though.
